packages/core/src/utils/requestQueue.ts

```diff
@@ -105,9 +109,30 @@ export const createRequestQueue = ({
           return logs;
         }
 
-        if (shouldRetry(error) === false || i === 3) throw error;
+        if (shouldRetry(error) === false) {
+          common.logger.warn({
+            service: "sync",
+            msg: `Failed '${request.method}' RPC request with non-retryable error: ${error}`,
+          });
+          throw error;
+        }
+
+        if (i === RETRY_COUNT) {
+          common.logger.warn({
+            service: "sync",
+            msg: `Failed '${request.method}' RPC request after ${
+              i + 1
+            } attempts with error: ${error}`,
```

Comment:
Let's be thoughtful about how to include errors within logs. 

I believe this will call `toString` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/toString#description) on the error object, but elsewhere we do `Error: ${error.message}`, and other times we omit the `with error` or `error:` entirely.

Not a solution here, just drawing attention.

Reply:
We should use `.message` in most places (wherever we don't want the stack trace)